tests/scripts/audit-validity-dates.py

```diff
@@ -0,0 +1,464 @@
+#!/usr/bin/env python3
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Audit validity date of X509 crt/crl/csr.
+
+This script is used to audit the validity date of crt/crl/csr used for testing.
+It would print the information about X.509 data if the validity period of the
+X.509 data didn't cover the provided validity period. The data are collected
+from tests/data_files/ and tests/suites/*.data files by default.
+"""
+
+import os
+import sys
+import re
+import typing
+import argparse
+import datetime
+import glob
+import logging
+from enum import Enum
+
+# The script requires cryptography >= 35.0.0 which is only available
+# for Python >= 3.6. Disable the pylint error here until we were
+# using modern system on our CI.
+from cryptography import x509 #pylint: disable=import-error
+
```

Comment:
When running the script manually with a version of `cryptography` that isn't ultra-recent (e.g. the version in the current Ubuntu LTS), you get a rather mysterious error. If you don't know what's going on, you have to find this comment to understand that you need to upgrade cryptography. And this comment is fragile: we might remove it when we no longer have Python 3.5 on our CI. I'd like to make this more robust and discovable. How about simply raising an error if `cryptography` is too old? Something like this:

```
import cryptography
if int(re.match(r'^[0-9]*', cryptography.__version__)[0]) < 35:
    raise Exception("audit-validity-dates requires cryptography >= 35.0.0 ({} is too old)".format(cryptography.__version__))
```

([You need a third-party package to properly compare version strings](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python). Let's not require a third-party packge.)